sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -787,9 +787,9 @@ public boolean advance() throws IOException {
       }
 
       /**
-       * Many objects used by Beam are mutable, but the Hadoop InputFormats tend to re-use the same
-       * object when returning them. Hence mutable objects returned by Hadoop InputFormats are
-       * cloned.
+       * Beam expects immutable objects, but the Hadoop InputFormats tend to re-use the same object
+       * when returning them. Hence mutable objects returned by Hadoop InputFormats are cloned., but
```

Comment:
There's a period followed by a comma, which isn't right. 

I'm not sure the "but the Hadoop InputFormats tend to re-use the same" adds much (it's just a repeat of a phrase in the last sentence)